apps/hash-external-services/docker-compose.yml

```diff
@@ -256,216 +164,34 @@ services:
         condition: service_healthy
       kratos-migrate:
         condition: service_completed_successfully
-      telemetry-collector:
-        condition: service_healthy
-    ports:
-      - "4433:4433" # public
-      - "4434:4434" # admin
     restart: unless-stopped
     environment:
       SECRETS_COOKIE: "${KRATOS_SECRETS_COOKIE}"
       SECRETS_CIPHER: "${KRATOS_SECRETS_CIPHER}"
       COURIER_SMTP_CONNECTION_URI: "smtps://test:test@mailslurper:1025/?skip_ssl_verify=true"
-      DSN: "postgres://${HASH_KRATOS_PG_USER}:${HASH_KRATOS_PG_PASSWORD}@postgres:${POSTGRES_PORT}/${HASH_KRATOS_PG_DEV_DATABASE}"
-      TRACING_PROVIDER: "jaeger"
-      TRACING_PROVIDERS_JAEGER_SAMPLING_SERVER_URL: "telemetry-collector:5778/sampling"
-      TRACING_PROVIDERS_JAEGER_LOCAL_AGENT_ADDRESS: "telemetry-collector:6831"
-      LOG_LEVEL: trace
+      DSN: "postgres://${HASH_KRATOS_PG_USER}:${HASH_KRATOS_PG_PASSWORD}@postgres:${POSTGRES_PORT}/${HASH_KRATOS_PG_DATABASE}"
     command: serve --dev --watch-courier
+    read_only: true
+    security_opt:
+      - no-new-privileges:true
     extra_hosts:
       - host.docker.internal:host-gateway
 
   mailslurper:
     image: oryd/mailslurper:latest-smtps
-    ports:
-      - "4436:4436"
-      - "4437:4437"
-
-  temporal-migrate:
-    build:
-      context: ./temporal
-      dockerfile: migrate.Dockerfile
-      args:
-        TEMPORAL_VERSION: "${HASH_TEMPORAL_VERSION}"
-    read_only: true
-    depends_on:
-      postgres:
-        condition: service_healthy
-    environment:
-      # This sets configuration values in
-      # https://github.com/temporalio/temporal/blob/master/docker/config_template.yaml
-      # posgres12 for v12+ of postgres.
-      DB: "postgres12"
-      DBNAME: "${HASH_TEMPORAL_PG_DEV_DATABASE}"
-      VISIBILITY_DBNAME: "${HASH_TEMPORAL_VISIBILITY_PG_DEV_DATABASE}"
-      DB_PORT: "5432"
-      # Intentionally use the POSTGRES user as it's the "superadmin" which has access to schema
-      POSTGRES_USER: "${POSTGRES_USER}"
-      POSTGRES_PWD: "${POSTGRES_PASSWORD}"
-      POSTGRES_SEEDS: "postgres" # the hostname of the postgres container
     security_opt:
       - no-new-privileges:true
 
-  temporal:
-    container_name: temporal
-    image: "temporalio/server:${HASH_TEMPORAL_VERSION}"
+  redis:
+    image: redis:6.2
+    read_only: true
```

Comment:
what does `read_only` do here? I thought it only applies to volumes?

Reply:
No, `read_only` applies to the container itself

Reply:
Note, that the `.prod` file specified this before as well